static/src/javascripts/projects/common/modules/search-tool.js

```diff
@@ -0,0 +1,197 @@
+                    return;
+                }
+
+                this.fetchData();
+            },
+
+            fetchData: function () {
+                return ajax({
+                    url: apiUrl + newQuery,
+                    type: 'json',
+                    crossOrigin: true
+                }).then(function (positions) {
+                    this.renderList(positions, 5);
+                    oldQuery = newQuery;
+                }.bind(this));
+            },
+
+            handleKeyEvents: function (e) {
+                var key = keyCodeMap[e.which || e.keyCode];
+
+                // Run this function only if we are inside the input
+                if (!$(e.target).hasClass('js-search-tool-input')) {
+                    return;
+                }
+
+                if (key === 'down') { // down
+                    e.preventDefault();
+                    this.move(1);
+                } else if (key === 'up') { // up
+                    e.preventDefault();
+                    this.move(-1);
+                } else if (key === 'enter') { // enter
+                    this.pushData();
+                } else {
+                    this.getListOfResults(e);
+                }
+            },
+
+            move: function (increment) {
+                var $active = $('.active', $list),
+                    id      = parseInt($active.attr('id'), 10);
+
+                if (isNaN(id)) {
+                    id = -1;
+                }
+
+                $active.removeClass('active');
+
+                // When outside of the list show latest query
+                if (this.getNewId(id + increment) < 0) {
+                    this.setInputValue(oldQuery);
+
+                    // When looping inside of the list show list item
+                } else {
+                    $('#' + this.getNewId(id + increment) + 'sti', $list).addClass('active');
+                    this.setInputValue();
+                }
+            },
+
+            getNewId: function (id) {
+                var len   = $('li', $list).length,
+                    newId = id % len;
+
+                // Make sure that we can hit saved input option which has position -1
+                if (newId < -1) {
+                    newId = len - 1;
+                } else if (id === len) {
+                    newId = -1;
+                }
+
+                return newId;
+            },
+
+            setInputValue: function (value) {
+                var inputValue = value || $('.active', $list).attr('data-weather-city');
+
+                $input.val(inputValue);
+            },
+
+            renderList: function (results, numOfResults) {
+                var docFragment   = document.createDocumentFragment(),
+                    resultsToShow = results.length - numOfResults;
+
+                _(results).initial(resultsToShow).each(function (item, index) {
+                    var li = document.createElement('li');
```

Comment:
there's a `$.create` method, will return a bonzo object to make the next two lines easier. think you have to supply html to it though, e.g.

```var $li = $.create('<li></li>');```